darts/timeseries.py

```diff
@@ -300,35 +837,100 @@ def gaps(self) -> pd.DataFrame:
-        ----------
-        deep
-            Make a deep copy. If False, the underlying pandas DataFrame will be the same
-
         Returns
         -------
         TimeSeries
             A copy of this time series.
         """
-        if deep:
-            return TimeSeries(self.pd_dataframe(), self.freq_str())
+        return TimeSeries(self._xa)  # the xarray will be copied in the TimeSeries constructor
+
+    def get_index_at_point(self, point: Union[pd.Timestamp, float, int], after=True) -> int:
+        """
+        Converts a point into an integer index
+
+        Parameters
+        ----------
+        point
+            This parameter supports 3 different data types: `pd.Timestamp`, `float` and `int`.
+
+            `pd.Timestamp` work only on series that are indexed with a `pd.DatetimeIndex`. In such cases, the returned
+            point will be the index of this timestamp if it is present in the series time index. It it's not present
+            in the time index, the index of the next timestamp is returned if `after=True` (if it exists in the series),
+            otherwise the index of the previous timestamp is returned (if it exists in the series).
+
+            In case of a `float`, the parameter will be treated as the proportion of the time series
+            that should lie before the point.
+
+            In the case of `int`, the parameter will returned as such, provided that it is in the series. Otherwise
+            it will raise a ValueError.
+        after
+            If the provided pandas Timestamp is not in the time series index, whether to return the index of the
+            next timestamp or the index of the previous one.
+
+        """
+        point_index = -1
+        if isinstance(point, float):
+            raise_if_not(0. <= point <= 1., 'point (float) should be between 0.0 and 1.0.', logger)
+            point_index = int((self.__len__() - 1) * point)
+        elif isinstance(point, int):
+            raise_if(point not in range(self.__len__()), "point (int) should be a valid index in series", logger)
+            point_index = point
+        elif isinstance(point, pd.Timestamp):
+            raise_if_not(self._has_datetime_index,
+                         'A Timestamp has been provided, but this series is not time-indexed.', logger)
+            self._raise_if_not_within(point)
+            if point in self:
+                point_index = self._time_index.get_loc(point)
+            else:
+                point_index = self._time_index.get_loc(self._get_first_timestamp_after(point) if after else
+                                                       self._get_last_timestamp_before(point))
         else:
-            return TimeSeries(self._df, self.freq_str())
+            raise_log(TypeError("`point` needs to be either `float`, `int` or `pd.Timestamp`"), logger)
+        return point_index
+
+    def get_timestamp_at_point(self, point: Union[pd.Timestamp, float, int]) -> pd.Timestamp:
```

Comment:
Wouldn't make sense to pass a Timestamp to get a timestamp, right?

Reply:
It could be used to check that the timestamp is indeed in the series. I'm a little bit unwilling to remove this in case it's relied on by another piece of logic somewhere... it doesn't really hurt I think.